packages/mobile/src/screens/sign-up-screen/components/layout.tsx

```diff
@@ -0,0 +1,74 @@
+import { Children } from 'react'
+
+import { type NativeBoxProps } from '@audius/harmony-native'
+import { Flex } from 'app/harmony-native/components/layout/Flex/Flex'
+import type { NativeFlexProps } from 'app/harmony-native/components/layout/Flex/types'
+import { Text } from 'app/harmony-native/foundations/typography/Text'
+
+type PageProps = NativeFlexProps & {
+  centered?: boolean
+}
+
+export const Page = (props: PageProps) => {
+  const { centered, children, ...other } = props
+
+  const childrenArray = Children.toArray(children)
+  const footer = childrenArray.pop()
+
+  const layoutProps: NativeFlexProps = {
+    direction: 'column',
+    h: '100%',
+    gap: '2xl',
+    ph: 'l',
+    pv: '2xl'
+  }
+
+  if (centered) {
+    return (
+      <Flex h='100%' direction='column' alignItems='center' {...other}>
+        <Flex {...layoutProps} alignSelf='center'>
+          {childrenArray}
+        </Flex>
+        {footer}
+      </Flex>
+    )
+  }
+
+  return (
+    <Flex {...layoutProps} {...other}>
+      {children}
+    </Flex>
+  )
+}
+
+type HeadingProps = {
+  prefix?: any
+  postfix?: any
+  heading: any
+  description?: any
+  centered?: boolean
+} & Omit<NativeFlexProps & NativeBoxProps, 'prefix'>
+
+export const Heading = (props: HeadingProps) => {
```

Comment:
cool how so much of this just works, btw thinking we should have "flexprops" in native with no reference. is the fear that folks will accidently import them from `@audius/harmony`? i think we shoud add an eslint restricted import rule for @audius/harmony since it should never be referenced except in the setup harmony-native code.

Reply:
Yeah the `FlexProps` comment is far fair. I was on the fence but originally left them separate but yeah I think after using everything a bit more its better if everything is just 1:1.
I like the eslint rule idea! That will force us to re-export necessary stuff through `harmony-native`